group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupConfig.java

```diff
@@ -238,4 +264,39 @@ public int shareHeartbeatIntervalMs() {
     public int shareRecordLockDurationMs() {
         return shareRecordLockDurationMs;
     }
+
+    /**
+     * The share group auto offset reset strategy.
+     */
+    public ShareGroupAutoOffsetReset shareAutoOffsetReset() {
+        return ShareGroupAutoOffsetReset.forStrategy(shareAutoOffsetReset);
+    }
+
+    public enum ShareGroupAutoOffsetReset {
```

Comment:
Why not to use existing `OffsetResetStrategy`?

Reply:
I would keep them separate so we can evolve them separately.

Reply:
Thanks for the review. It's better to use OffsetResetStrategy, so I'll remove the newly added enum.

Reply:
> It's better to use OffsetResetStrategy, so I'll remove the newly added enum.

Did you miss it or changed mind?

What's the final decision, are we keeping it? I would prefer to use `OffsetResetStrategy` as don't see any difference to one defined as new. If we need to have an independent one, in future, because of some more states then create a new one. Else it's unnecessary.

Reply:
I expect that `OffsetResetStrategy` will evolve independently and I prefer to separate them now.

Reply:
Had a word with @AndrewJSchofield. Sure, let's have separate one.

@chirag-wadhwa5 When we define just earliest and latest in this enum then do we need the validations we have specified for this config?

Reply:
I don't think we would require any explicit validations on top of it. I have removed the unnecessary validations now.